device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/CorrelatingMessageCallback.java

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package com.microsoft.azure.sdk.iot.device;
+
+import com.microsoft.azure.sdk.iot.device.transport.IotHubTransportPacket;
+
+/**
+ * This interface has a number of call backs to notify when a message is in a specific part of the {@link com.microsoft.azure.sdk.iot.device.transport.IotHubTransport} lifecycle.
+ * <p>
+ *     There is no default implementation of this interface and will be up to the developer to implement.
+ * </p>
+ * <p>
+ *     An example of this implementation and usage can be found in the DeviceTwinSample.
+ * </p>
+ */
+public interface CorrelatingMessageCallback
```

Comment:
Considering this callback should only ever be used for tracking an outgoing twin message, should this name be a bit more specific? Something like "ReportedPropertyMessageStateCallback" or something?

Reply:
It could be used with a regular Message class in sendEvent if the customer sets it for the base message.